Assets/MixedRealityToolkit.Extensions/SpectatorView/Scripts/Editor/CompositorWindow.cs

```diff
@@ -0,0 +1,346 @@
+            if (hologramSettingsFoldout)
+            {
+
+                EditorGUILayout.BeginVertical("Box");
+                {
+                    CompositionManager compositionManager = GetCompositionManager();
+
+                    if (compositionManager != null)
+                    {
+                        EditorGUILayout.Space();
+
+                        GUI.enabled = compositionManager == null || !compositionManager.IsVideoFrameProviderInitialized;
+                        compositionManager.CaptureDevice = (CompositionManager.FrameProviderDeviceType)EditorGUILayout.Popup("Video source", ((int)compositionManager.CaptureDevice), Enum.GetNames(typeof(CompositionManager.FrameProviderDeviceType)));
+                        GUI.enabled = true;
+
+                        EditorGUILayout.Space();
+                    }
+
+                    float newAlpha = EditorGUILayout.Slider("Alpha", hologramAlpha, 0, 1);
+                    if (newAlpha != hologramAlpha)
+                    {
+                        hologramAlpha = newAlpha;
+                        if (compositionManager != null && compositionManager.TextureManager != null)
+                        {
+                            compositionManager.TextureManager.SetHologramShaderAlpha(newAlpha);
+                        }
+                    }
+
+                    EditorGUILayout.Space();
+
+                    if (compositionManager != null)
+                    {
+                        EditorGUILayout.BeginHorizontal();
+                        {
+                            float previousFrameOffset = compositionManager.VideoTimestampToHolographicTimestampOffset;
+                            compositionManager.VideoTimestampToHolographicTimestampOffset = EditorGUILayout.Slider("Frame time adjustment", previousFrameOffset, -1 * maxFrameOffset, maxFrameOffset);
+                        }
+                        EditorGUILayout.EndHorizontal();
+                    }
+
+                    EditorGUILayout.Space();
+                }
+                EditorGUILayout.EndVertical();
+            }
+        }
+
+        private void CompositorStatsGUI()
+        {
+            compositorStatsFoldout = EditorGUILayout.Foldout(compositorStatsFoldout, "Compositor Stats");
+            if (compositorStatsFoldout)
+            {
+                CompositionManager compositionManager = GetCompositionManager();
+                if (compositionManager != null && compositionManager.IsVideoFrameProviderInitialized)
+                {
+                    UpdateStatistics(compositionManager);
+
+                    RenderTitle(framerateStatisticsMessage, framerateStatisticsColor);
+                    
+                    int queuedFrameCount = compositionManager.GetQueuedOutputFrameCount();
+                    Color queuedFrameColor = (queuedFrameCount > 6) ? Color.green : Color.red;
```

Comment:
>6 [](start = 65, length = 1)

What's the significance of the number 6?

Reply:
At 60fps video, 6 represents only 100ms of remaining buffer (out of the 300ms worth of frames the compositor buffers, which should remain full as long as the Unity scene is running at 60fps or higher). The value is arbitrary, but we wanted anything out of the ordinary to be immediately flagged as red when working on stage demos so that it's easy to glance at the compositor window and see if something is wrong.